src/evotorch/decorators.py

```
@@ -694,6 +950,12 @@ def f(a: torch.Tensor, b: torch.Tensor) -> torch.Tensor: ...
     )
 
     def expects_ndim_decorator(fn: Callable):
+        if hasattr(fn, "__evotorch_distribute__"):
+            raise ValueError(
+                "Cannot apply `@expects_ndim` or `@rowwise` on a function"
+                " that was previously subjected to `@distribute`"
+            )
```

Ruff (0.14.0)

Avoid specifying long messages outside the exception class

(TRY003)